src/compat/array/takeWhile.ts

```diff
@@ -0,0 +1,157 @@
+ *
+ * If the array is `null` or `undefined`, returns an empty array.
+ *
+ * @template T
+ * @param {ArrayLike<T> | null | undefined} array - The array to process.
+ * @param {PropertyKey} property - A property key. Elements are included if they have a truthy value for this key.
+ * @returns {T[]} - A slice of the array with elements taken from the beginning or an empty array if `array` is `null` or `undefined`.
+ *
+ * @example
+ * const items = [{ valid: true }, { valid: true }, { valid: false }];
+ * const result = takeWhile(items, 'valid');
+ * console.log(result); // [{ valid: true }, { valid: true }]
+ */
+export function takeWhile<T>(array: ArrayLike<T> | null | undefined, property: PropertyKey): T[];
+
+/**
+ * Creates a slice of the array with elements taken from the beginning while the specified predicate is satisfied.
+ * If no predicate is provided, the identity function is used by default.
+ * If the array is `null` or `undefined`, returns an empty array.
+ *
+ * @template T
+ * @param {ArrayLike<T> | null | undefined} array - The array to process.
+ * @param {(item: T, index: number, array: T[]) => unknown | Partial<T> | [keyof T, unknown] | PropertyKey} [predicate] - The condition used to determine elements to include. Can be:
+ * - A function invoked per iteration.
+ * - A partial object to match properties.
+ * - A key-value pair as a tuple.
+ * - A property key to check for truthy values.
+ * Defaults to the identity function if not provided.
+ * @returns {T[]} - A slice of the array with elements taken from the beginning or an empty array if `array` is `null` or `undefined`.
+ *
+ * @example
+ * // Using a predicate function
+ * const items = [1, 2, 3, 4, 5];
+ * const result = takeWhile(items, (item) => item < 3);
+ * console.log(result); // [1, 2]
+ *
+ * // Using a partial object
+ * const items2 = [{ id: 30 }, { id: 20 }, { id: 10 }];
+ * const result2 = takeWhile(items2, { id: 30 });
+ * console.log(result2); // [{ id: 30 }]
+ *
+ * // Using a key-value pair
+ * const items3 = [{ name: 'Alice' }, { name: 'Bob' }, { name: 'Alice' }];
+ * const result3 = takeWhile(items3, ['name', 'Alice']);
+ * console.log(result3); // [{ name: 'Alice' }]
+ *
+ * // Using a property key
+ * const items4 = [{ active: true }, { active: true }, { active: false }];
+ * const result4 = takeWhile(items4, 'active');
+ * console.log(result4); // [{ active: true }, { active: true }]
+ *
+ * // No predicate provided
+ * const items5 = [true, false];
+ * const result5 = takeWhile(items5);
+ * console.log(result5); // [true]
+ *
+ * // null or undefined array
+ * const result6 = takeWhile(null);
+ * console.log(result6); // []
+ **/
+export function takeWhile<T>(
+  _array: ArrayLike<T> | null | undefined,
+  predicate?:
+    | ((value: T, index: number, array: ArrayLike<T>) => unknown)
+    | Partial<T>
+    | [keyof T, unknown]
+    | PropertyKey
+): T[] {
+  if (!isArrayLikeObject(_array)) {
+    return [];
+  }
+
+  const array = toArray(_array);
```

Comment:
```suggestion
  array: ArrayLike<T> | null | undefined,
  predicate?:
    | ((value: T, index: number, array: ArrayLike<T>) => unknown)
    | Partial<T>
    | [keyof T, unknown]
    | PropertyKey
): T[] {
  if (!isArrayLikeObject(_array)) {
    return [];
  }

  array = toArray(array);
```

Since it's a function parameter and exposed externally, how about writing it like this instead of including an underscore?

Reply:
I also like your approach.

However, I wrote it this way to maintain consistency with the existing code.
Since `compat/takeRightWhile` uses underscore, I followed the same convention here for consistency.

Do you think it would be better to make the change as suggested?

Reply:
Only this function is written that way, while most of the other functions don't use underscores.

There are occasionally functions that don't follow the convention, so when implementing a compat function, it's a good idea to also refer to Lodash's interface!

Reply:
Thank you so much for carefully submitting the PR while referring to other code!

Reply:
I removed underscores from the parameter name and refactored the implementation to use internal variables instead.

Please let me know if there's a cleaner or more idiomatic way to approach this.